cmd/cody-gateway/internal/httpapi/completions/fireworks.go

```diff
@@ -70,10 +71,9 @@ func NewFireworksHandler(
 					}
 				}
 			},
-			getRequestMetadata: func(ctx context.Context, logger log.Logger, act *actor.Actor, body fireworksRequest) (model string, additionalMetadata map[string]any) {
-				// we checked this is a code completion request in validateRequest
-				// check that actor is a PLG user
-				if logSelfServeCodeCompletionRequests && act.IsDotComActor() {
+			getRequestMetadata: func(ctx context.Context, logger log.Logger, act *actor.Actor, feature codygateway.Feature, body fireworksRequest) (model string, additionalMetadata map[string]any) {
+				// Check that this is a code completion request and that the actor is a PLG user
+				if feature != codygateway.FeatureCodeCompletions && logSelfServeCodeCompletionRequests && act.IsDotComActor() {
```

Comment:
```suggestion
				if feature != codygateway.FeatureCodeCompletions && logSelfServeCodeCompletionRequests && act.IsDotComActor() {
```
Should this be ==? This code logs code completions, but != sould stop it from being called

Reply:
YES! 😰